packages/headless/src/features/search/search-actions.ts

```diff
@@ -342,7 +342,9 @@ const addEntryInActionsHistory = (state: StateNeededByExecuteSearch) => {
   if (state.configuration.analytics.enabled) {
     historyStore.addElement({
       name: 'Query',
-      value: state.query?.q || getQueryInitialState().q,
+      ...(state.query?.q && {
```

Comment:
We don't really have to do this anymore

Reply:
In this case, I would find it easier to read if we wrote the function like this:
```ts
const addEntryInActionsHistory = (state: StateNeededByExecuteSearch) => {
  if (state.configuration.analytics.enabled) {
    const element: TYPEHERE = {
      name: 'Query',
      time: JSON.stringify(new Date()),
    };
    if (state.query?.q) {
      element.value = state.query.q;
    }
    historyStore.addElement(element);
  }
};
```

Reply:
Or:

```
if (state.configuration.analytics.enabled && state.query.q) {
  historyStore.addElement({...})
}
```

Reply:
No we still need to send the Query even if the value is empty